### PR TITLE
Use project root as `T.workspace` also in meta-build

### LINKED ISSUE #2802: `semanticDbData` fails on `--meta-level 1`
Found this while investigating BSP and metals problems.
This happens when using Metals with Mill BSP.

You can reproduce the issue from Mill's CLI running:
```
mill --meta-level 1 semanticDbData
```
crashes with:
<details>
<summary>Click for stack trace</summary>

```
[info] compiling 1 Scala source to /Users/lorenzo/scala/repro/out/mill-build/semanticDbData.dest/classes ...
[error] failed to generate semanticdb for /Users/lorenzo/scala/repro/out/mill-build/generateScriptSources.dest/millbuild/build.sc:
[error] java.nio.file.NoSuchFileException: /Users/lorenzo/scala/repro/mill-build
[error]         at java.base/sun.nio.fs.UnixException.translateToIOException(UnixException.java:92)
[error]         at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:111)
[error]         at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:116)
[error]         at java.base/sun.nio.fs.UnixPath.toRealPath(UnixPath.java:860)
[error]         at scala.meta.internal.semanticdb.scalac.SemanticdbConfig.realSourceRoot$lzycompute(ConfigOps.scala:22)
[error]         at scala.meta.internal.semanticdb.scalac.SemanticdbConfig.realSourceRoot(ConfigOps.scala:22)
[error]         at scala.meta.internal.semanticdb.scalac.InputOps$ImplicitSemanticdbConfig$.$anonfun$fileInSourceRoot$extension$1(InputOps.scala:105)
[error]         at scala.Option.orElse(Option.scala:477)
[error]         at scala.meta.internal.semanticdb.scalac.InputOps$ImplicitSemanticdbConfig$.fileInSourceRoot$extension(InputOps.scala:104)
[error]         at scala.meta.internal.semanticdb.scalac.InputOps$XtensionGSourceFileInput.isInSourceroot(InputOps.scala:25)
[error]         at scala.meta.internal.semanticdb.scalac.SemanticdbPipeline$SemanticdbTyperComponent$ComputeSemanticdbPhase.saveSemanticdbForCompilationUnit(SemanticdbPipeline.scala:58)
[error]         at scala.meta.internal.semanticdb.scalac.SemanticdbPipeline$SemanticdbTyperComponent$ComputeSemanticdbPhase.apply(SemanticdbPipeline.scala:65)
[error]         at scala.tools.nsc.Global$GlobalPhase.applyPhase(Global.scala:467)
[error]         at scala.tools.nsc.Global$GlobalPhase.run(Global.scala:414)
[error]         at scala.meta.internal.semanticdb.scalac.SemanticdbPipeline$SemanticdbTyperComponent$ComputeSemanticdbPhase.run(SemanticdbPipeline.scala:75)
[error]         at scala.tools.nsc.Global$Run.compileUnitsInternal(Global.scala:1530)
[error]         at scala.tools.nsc.Global$Run.compileUnits(Global.scala:1514)
[error]         at scala.tools.nsc.Global$Run.compileSources(Global.scala:1506)
[error]         at scala.tools.nsc.Global$Run.compileFiles(Global.scala:1619)
[error]         at xsbt.CachedCompiler0.run(CompilerBridge.scala:163)
[error]         at xsbt.CachedCompiler0.run(CompilerBridge.scala:134)
[error]         at xsbt.CompilerBridge.run(CompilerBridge.scala:39)
[error]         at sbt.internal.inc.AnalyzingCompiler.compile(AnalyzingCompiler.scala:91)
[error]         at sbt.internal.inc.MixedAnalyzingCompiler.$anonfun$compile$7(MixedAnalyzingCompiler.scala:193)
[error]         at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.scala:18)
[error]         at sbt.internal.inc.MixedAnalyzingCompiler.timed(MixedAnalyzingCompiler.scala:248)
[error]         at sbt.internal.inc.MixedAnalyzingCompiler.$anonfun$compile$4(MixedAnalyzingCompiler.scala:183)
[error]         at sbt.internal.inc.MixedAnalyzingCompiler.$anonfun$compile$4$adapted(MixedAnalyzingCompiler.scala:163)
[error]         at sbt.internal.inc.JarUtils$.withPreviousJar(JarUtils.scala:239)
[error]         at sbt.internal.inc.MixedAnalyzingCompiler.compileScala$1(MixedAnalyzingCompiler.scala:163)
[error]         at sbt.internal.inc.MixedAnalyzingCompiler.compile(MixedAnalyzingCompiler.scala:211)
[error]         at sbt.internal.inc.IncrementalCompilerImpl.$anonfun$compileInternal$1(IncrementalCompilerImpl.scala:534)
[error]         at sbt.internal.inc.IncrementalCompilerImpl.$anonfun$compileInternal$1$adapted(IncrementalCompilerImpl.scala:534)
[error]         at sbt.internal.inc.Incremental$.$anonfun$apply$5(Incremental.scala:180)
[error]         at sbt.internal.inc.Incremental$.$anonfun$apply$5$adapted(Incremental.scala:178)
[error]         at sbt.internal.inc.Incremental$$anon$2.run(Incremental.scala:464)
[error]         at sbt.internal.inc.IncrementalCommon$CycleState.next(IncrementalCommon.scala:116)
[error]         at sbt.internal.inc.IncrementalCommon$$anon$1.next(IncrementalCommon.scala:56)
[error]         at sbt.internal.inc.IncrementalCommon$$anon$1.next(IncrementalCommon.scala:52)
[error]         at sbt.internal.inc.IncrementalCommon.cycle(IncrementalCommon.scala:263)
[error]         at sbt.internal.inc.Incremental$.$anonfun$incrementalCompile$8(Incremental.scala:419)
[error]         at sbt.internal.inc.Incremental$.withClassfileManager(Incremental.scala:506)
[error]         at sbt.internal.inc.Incremental$.incrementalCompile(Incremental.scala:406)
[error]         at sbt.internal.inc.Incremental$.apply(Incremental.scala:172)
[error]         at sbt.internal.inc.IncrementalCompilerImpl.compileInternal(IncrementalCompilerImpl.scala:534)
[error]         at sbt.internal.inc.IncrementalCompilerImpl.$anonfun$compileIncrementally$1(IncrementalCompilerImpl.scala:488)
[error]         at sbt.internal.inc.IncrementalCompilerImpl.handleCompilationError(IncrementalCompilerImpl.scala:332)
[error]         at sbt.internal.inc.IncrementalCompilerImpl.compileIncrementally(IncrementalCompilerImpl.scala:425)
[error]         at sbt.internal.inc.IncrementalCompilerImpl.compile(IncrementalCompilerImpl.scala:137)
[error]         at mill.scalalib.worker.ZincWorkerImpl.compileInternal(ZincWorkerImpl.scala:554)
[error]         at mill.scalalib.worker.ZincWorkerImpl.$anonfun$compileMixed$1(ZincWorkerImpl.scala:332)
[error]         at mill.api.FixSizedCache.withCachedValue(FixSizedCache.scala:66)
[error]         at mill.scalalib.worker.ZincWorkerImpl.withCompilers(ZincWorkerImpl.scala:411)
[error]         at mill.scalalib.worker.ZincWorkerImpl.compileMixed(ZincWorkerImpl.scala:331)
[error]         at mill.scalalib.ScalaModule.$anonfun$semanticDbData$2(ScalaModule.scala:548)
[error]         at mill.define.Task$TraverseCtx.evaluate(Task.scala:71)
[error]         at mill.eval.GroupEvaluator.$anonfun$evaluateGroup$11(GroupEvaluator.scala:308)
[error]         at scala.util.DynamicVariable.withValue(DynamicVariable.scala:59)
[error]         at scala.Console$.withErr(Console.scala:193)
[error]         at mill.api.SystemStreams$.$anonfun$withStreams$2(SystemStreams.scala:62)
[error]         at scala.util.DynamicVariable.withValue(DynamicVariable.scala:59)
[error]         at scala.Console$.withOut(Console.scala:164)
[error]         at mill.api.SystemStreams$.$anonfun$withStreams$1(SystemStreams.scala:61)
[error]         at scala.util.DynamicVariable.withValue(DynamicVariable.scala:59)
[error]         at scala.Console$.withIn(Console.scala:227)
[error]         at mill.api.SystemStreams$.withStreams(SystemStreams.scala:60)
[error]         at mill.eval.GroupEvaluator.$anonfun$evaluateGroup$8(GroupEvaluator.scala:308)
[error]         at mill.eval.GroupEvaluator.$anonfun$evaluateGroup$8$adapted(GroupEvaluator.scala:277)
[error]         at scala.collection.immutable.Vector.foreach(Vector.scala:2122)
[error]         at mill.eval.GroupEvaluator.computeAll$1(GroupEvaluator.scala:277)
[error]         at mill.eval.GroupEvaluator.evaluateGroup(GroupEvaluator.scala:332)
[error]         at mill.eval.GroupEvaluator.$anonfun$evaluateGroupCached$20(GroupEvaluator.scala:201)
[error]         at scala.util.DynamicVariable.withValue(DynamicVariable.scala:59)
[error]         at mill.eval.GroupEvaluator.evaluateGroupCached(GroupEvaluator.scala:192)
[error]         at mill.eval.GroupEvaluator.evaluateGroupCached$(GroupEvaluator.scala:43)
[error]         at mill.eval.EvaluatorImpl.evaluateGroupCached(EvaluatorImpl.scala:15)
[error]         at mill.eval.EvaluatorCore.$anonfun$evaluate0$2(EvaluatorCore.scala:116)
[error]         at scala.concurrent.impl.Promise$Transformation.run(Promise.scala:467)
[error]         at mill.eval.ExecutionContexts$RunNow$.execute(ExecutionContexts.scala:13)
[error]         at scala.concurrent.impl.Promise$Transformation.submitWithValue(Promise.scala:429)
[error]         at scala.concurrent.impl.Promise$DefaultPromise.submitWithValue(Promise.scala:338)
[error]         at scala.concurrent.impl.Promise$DefaultPromise.dispatchOrAddCallbacks(Promise.scala:312)
[error]         at scala.concurrent.impl.Promise$DefaultPromise.map(Promise.scala:182)
[error]         at mill.eval.EvaluatorCore.$anonfun$evaluate0$1(EvaluatorCore.scala:92)
[error]         at mill.eval.EvaluatorCore.$anonfun$evaluate0$1$adapted(EvaluatorCore.scala:90)
[error]         at scala.collection.immutable.Vector.foreach(Vector.scala:2122)
[error]         at mill.eval.EvaluatorCore.evaluate0(EvaluatorCore.scala:90)
[error]         at mill.eval.EvaluatorCore.$anonfun$evaluate$1(EvaluatorCore.scala:43)
[error]         at scala.util.DynamicVariable.withValue(DynamicVariable.scala:59)
[error]         at mill.eval.EvaluatorCore.evaluate(EvaluatorCore.scala:34)
[error]         at mill.eval.EvaluatorCore.evaluate$(EvaluatorCore.scala:26)
[error]         at mill.eval.EvaluatorImpl.evaluate(EvaluatorImpl.scala:15)
[error]         at mill.main.RunScript$.evaluateNamed(RunScript.scala:38)
[error]         at mill.main.RunScript$.$anonfun$evaluateTasksNamed$2(RunScript.scala:26)
[error]         at scala.util.Either.map(Either.scala:382)
[error]         at mill.main.RunScript$.evaluateTasksNamed(RunScript.scala:26)
[error]         at mill.runner.MillBuildBootstrap$.evaluateWithWatches(MillBuildBootstrap.scala:397)
[error]         at mill.runner.MillBuildBootstrap.$anonfun$processFinalTargets$3(MillBuildBootstrap.scala:307)
[error]         at scala.util.DynamicVariable.withValue(DynamicVariable.scala:59)
[error]         at mill.runner.MillBuildBootstrap.processFinalTargets(MillBuildBootstrap.scala:307)
[error]         at mill.runner.MillBuildBootstrap.evaluateRec(MillBuildBootstrap.scala:188)
[error]         at mill.runner.MillBuildBootstrap.state$lzycompute$1(MillBuildBootstrap.scala:76)
[error]         at mill.runner.MillBuildBootstrap.state$1(MillBuildBootstrap.scala:76)
[error]         at mill.runner.MillBuildBootstrap.evaluateRec(MillBuildBootstrap.scala:77)
[error]         at mill.runner.MillBuildBootstrap.evaluate(MillBuildBootstrap.scala:47)
[error]         at mill.runner.MillMain$.$anonfun$main0$6(MillMain.scala:213)
[error]         at mill.runner.Watching$.watchLoop(Watching.scala:27)
[error]         at mill.runner.MillMain$.$anonfun$main0$1(MillMain.scala:198)
[error]         at scala.util.DynamicVariable.withValue(DynamicVariable.scala:59)
[error]         at scala.Console$.withErr(Console.scala:193)
[error]         at mill.api.SystemStreams$.$anonfun$withStreams$2(SystemStreams.scala:62)
[error]         at scala.util.DynamicVariable.withValue(DynamicVariable.scala:59)
[error]         at scala.Console$.withOut(Console.scala:164)
[error]         at mill.api.SystemStreams$.$anonfun$withStreams$1(SystemStreams.scala:61)
[error]         at scala.util.DynamicVariable.withValue(DynamicVariable.scala:59)
[error]         at scala.Console$.withIn(Console.scala:227)
[error]         at mill.api.SystemStreams$.withStreams(SystemStreams.scala:60)
[error]         at mill.runner.MillMain$.main0(MillMain.scala:80)
[error]         at mill.runner.MillServerMain$.main0(MillServerMain.scala:83)
[error]         at mill.runner.MillServerMain$.main0(MillServerMain.scala:35)
[error]         at mill.runner.Server.$anonfun$handleRun$1(MillServerMain.scala:186)
[error]         at java.base/java.lang.Thread.run(Thread.java:829)
[error] one error found
```

</details>

The reason is that `T.workspace` when invoked the `--meta-level 1` returns the `./mill-build/` directory, which doesn't exist if you didn't set a create a meta-build.
We pass the directory to the `semanticdb` plugin [here](https://github.com/com-lihaoyi/mill/blob/a614c898e485f79de42b514fef3389d04dd96c19/scalalib/src/mill/scalalib/ScalaModule.scala#L556) which then fails since the directory doesn't exist.

Should `T.workspace` point to `./mill-build` when invoked in a different `meta-level`? In case it should, should we have another `def` to get the actual project root?

### PR DESCRIPTION
**Pull request by lolgab**
Fixes #2802

Pull request: https://github.com/com-lihaoyi/mill/pull/2809

### CODE REVIEW

File: main/eval/src/mill/eval/EvaluatorImpl.scala [hunk 1]
```diff
@@ -14,6 +14,7 @@ import scala.reflect.ClassTag
  */
 private[mill] case class EvaluatorImpl(
     home: os.Path,
+    projectRoot: os.Path,
```
Comment (lefou) [inline main/eval/src/mill/eval/EvaluatorImpl.scala:17]:
```suggestion
    workspaceRoot: os.Path,
```

We should try to use the same term to establish some domain language.

Comment (lolgab) [reply main/eval/src/mill/eval/EvaluatorImpl.scala:17]: Then I would just call it `workspace`. What do you think?

Comment (lefou) [reply main/eval/src/mill/eval/EvaluatorImpl.scala]: Sound good.

**Review by lefou — APPROVED** [review]
Looks good to me.

**Pull request merged**